page_iterator.go

```diff
@@ -108,30 +106,38 @@ func (pI *PageIterator) SetReqOptions(reqOptions []abstractions.RequestOption) {
 }
 
 func (pI *PageIterator) hasNext() bool {
-	if pI.currentPage == nil || pI.currentPage.getNextLink() == nil {
+	if pI.currentPage.getNextLink() == nil {
```

Comment:
```suggestion
	return pI.currentPage.getNextLink() != nil
```

and drop the rest, or drop the function all together since it's private and is not used anywhere.